main.go

```diff
@@ -522,29 +551,29 @@ func (p *OAuthProxy) setupRoutes(r *gin.Engine) {
 		MaxAge:           12 * time.Hour,
 	}))
 
-	// Health endpoint
+	// Health endpoint (no prefix)
 	r.GET("/health", p.healthHandler)
 
 	// Rate limiting middleware
 	if p.rateLimiter != nil {
 		r.Use(p.rateLimitMiddleware())
 	}
 
-	// OAuth endpoints
-	r.GET("/authorize", p.authorizeHandler)
-	r.POST("/authorize", p.authorizeHandler)
-	r.GET("/callback", p.callbackHandler)
-	r.POST("/token", p.tokenHandler)
-	r.POST("/revoke", p.revokeHandler)
-	r.POST("/register", p.registerHandler)
+	// OAuth endpoints (with prefix)
+	r.GET(addPrefix("/authorize"), p.authorizeHandler)
+	r.POST(addPrefix("/authorize"), p.authorizeHandler)
+	r.GET(addPrefix("/callback"), p.callbackHandler)
+	r.POST(addPrefix("/token"), p.tokenHandler)
+	r.POST(addPrefix("/revoke"), p.revokeHandler)
+	r.POST(addPrefix("/register"), p.registerHandler)
 
-	// Metadata endpoints
-	r.GET("/.well-known/oauth-authorization-server", p.oauthMetadataHandler)
-	r.GET("/.well-known/oauth-protected-resource/mcp", p.protectedResourceMetadataHandler)
+	// Metadata endpoints (with prefix)
+	r.GET(addPrefix("/.well-known/oauth-authorization-server"), p.oauthMetadataHandler)
+	r.GET(addPrefix("/.well-known/oauth-protected-resource/mcp"), p.protectedResourceMetadataHandler)
```

Comment:
This is where the MCP spec and the dependent RFCs get complicated. If we have a path prefix, say `prefix`, then this URL should be `/.well-known/oauth-protected-resource/prefix/mcp`.